### PR TITLE
[argp_standalone_jll] Build for experimental platforms

### PR DESCRIPTION
**Pull request by jpsamaroo**

### CODE REVIEW

File: A/argp_standalone/build_tarballs.jl [hunk 1]
```diff
@@ -36,4 +36,4 @@ dependencies = Dependency[
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```
Comment (giordano) [inline A/argp_standalone/build_tarballs.jl:39]: You can't restrict to v1.6 without changing the version number, it'll break the package for current users on v1.5-

**Comment by jpsamaroo** [comment]
It appears they only release as major+minor, so I used the patch version.

**Comment by jpsamaroo** [comment]
Crap, I forgot to expand the cxxstring ABIs... Is there a reason we don't do this by default in `supported_platforms()`?

**Comment by giordano** [comment]
You didn't have to, what's the problem? 🤔

**Comment by jpsamaroo** [comment]
https://github.com/JuliaPackaging/Yggdrasil/pull/3348 is still unable to find a matching argp_standalone for armv6l, which I assumed was due to the cxxstring stuff. But I guess not?

**Comment by giordano** [comment]
> Is there a reason we don't do this by default in supported_platforms()?

Because it isn't always needed, and if we enabled it by default that'd be a waste of resource and it'd be way harder to tell whether it isn't needed to expand the C++ string ABIs

**Pull request merged**